core/object/class_db.cpp

```diff
@@ -1073,6 +1073,7 @@ void ClassDB::get_property_list(const StringName &p_class, List<PropertyInfo> *p
 	ClassInfo *type = classes.getptr(p_class);
 	ClassInfo *check = type;
 	while (check) {
+		// Intentionally making a copy as we may modify it.
```

Comment:
Added a comment as making this a reference compiled fine, but actually lead to changes in the classref.